### ISSUE TITLE
[Feature] Allow resume at specified step

### ISSUE THREAD

**Issue by dreamflasher** [wandb/wandb#2234]
**Is your feature request related to a problem? Please describe.**
Runs crash randomly, we do checkpointing every 50 epochs, and currently it's not possible to specify that we want to resume at epoch 250. 

**Describe the solution you'd like**
Add epoch/step parameter to wandb.init(resume=True), that will override any future logged steps.

**Comment by tyomhak** [comment]
Hey @dreamflasher , thank you for the request. Unfortunately it's currently not possible, but it's definitely in our plans.

**Comment by ashok-arjun** [comment]
Hey @dreamflasher you can always specify the `step` in `wandb.log` as

`wandb.log(dict, step=iteration)`

You can save the iteration also in the model `.pth` file, and when you're retrieving a model to resume a run, you can also retrieve the `iteration` and set it in your code before starting any wandb logging

**Comment by wztdream** [comment]
Hi,
What is the status of this feature? It really helpful, now I have to restart a new run and the logged data will be in different places, which is very inconvenient.

I also found we can not use:

```
wandb.log(dict, step=iteration)
```
I tried it, and it shows the old value are not overwrite by the new value.

**Comment by kptkin** [comment]
closing in favor of #1395 as it a duplicate
